src/rax.c

```diff
@@ -192,6 +192,7 @@ rax *raxNew(void) {
     rax->numele = 0;
     rax->numnodes = 1;
     rax->head = raxNewNode(0, 0);
+    rax->alloc_size = rax_alloc_size(rax) + rax_alloc_size(rax->head);
```

Comment:
Good point, I had the same remark for myself while working on it. Not super fond of `rax_node_alloc_size` though because the name really replaces an allocator call, not one to a specific data type. Case in point, `rax_alloc_size(rax)` is on a `rax*` not a `raxNode*`.

I think I would prefer `rax_ptr_alloc_size`.

But this might not even be needed at all. Grabbing the size from the allocator is only useful in this PR. If I manage to track the memory entirely under rax_malloc, rax_realloc, rax_free, then we don't need rax_alloc_size at all. I'll try tomorrow if time permits.

Reply:
Never mind, that doesn't work. Basic idea was to have calls like `rax_malloc(rax, size)` so that rax_malloc would itself update rax->alloc_size (same with rax_realloc and rax_free).

It falls well short for example within raxNewNode() because...there's no rax* in that context. Looking back my idea was somewhat naive.

I've just now pushed a new commit with just s/rax_alloc_size/rax_ptr_alloc_size/. Let me know your thoughts. If that works what remains is plugging rax->alloc_size to the memory stats.

Reply:
`rax_ptr_alloc_size` is good.

Now I'd just like to have a test where we compare raxAllocSize to the allocator's view of the used memory, to check that we compute it correctly.

Reply:
> Now I'd just like to have a test where we compare raxAllocSize to the allocator's view of the used memory, to check that we compute it correctly.

I've added a commit to demo this. It simply asserts that raxAllocSize(rax) and zmalloc_used_memory() agree. I added one spot, but I am thinking adding it whenever we insert/delete an element in those tests. Then we should be pretty confident that the two are aligned.

(I also double checked that `TEST_ASSERT(raxAllocSize(t)+1== zmalloc_used_memory());` indeed fails the test)

Reply:
You can add it at a few places, such once after inserting many elements, once after deleting some, etc.

Reply:
Ok I have added a few more places.

In some tests I could not add because for example on is comparing rax against a hash table, and the hash table has an impact on zmalloc_used_memory too.